lib/pbench/server/database/alembic/versions/9cc638cb865c_operationname.py

```diff
@@ -0,0 +1,36 @@
+"""Correct operationname ENUM
+
+Alembic does not check or autogenerate ENUM value changes, so this will
+upgrade the original operationname revision to add UPDATE.
+
+Revision ID: 9cc638cb865c
+Revises: 9df060db17de
+Create Date: 2023-03-03 14:32:16.955897
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "9cc638cb865c"
+down_revision = "9df060db17de"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # NOTE: the 'BACKUP' and 'UNPACK' operationname ENUM values are now
+    # obsolete. We don't attempt to remove them because dataset_operations
+    # rows with these values record the previous operation of the server.
+    # Instead we simply add the new 'UPDATE' value. Having the obsolete
+    # values defined in the ENUM isn't a problem, and removing them isn't
+    # worth the complication.
+    with op.get_context().autocommit_block():
+        op.execute("ALTER TYPE operationname ADD VALUE 'UPDATE' BEFORE 'UPLOAD'")
```

Comment:
What does it mean to add a value _before_ another value?  Doesn't that effectively change the value of the second one?  It seems like that would create a state which was difficult to downgrade from (i.e., a state which would render the comment at lines 33-34 incorrect...).

If we're trying to maintain compatibility (e.g., by never removing old values), then it seems like new values would always need to be added _at the end_.

Reply:
A PostgreSQL ENUM is effectively a string column with a value check against a set of literal strings, so there's no hidden internal order-sensitive representation to break.

On the other hand, my brain spent the first part of my hike gnawing on something at the core of this comment, and I came to two conclusions (which allowed me to relax and enjoy the rest of the hike):

1. It's probably best to develop a convention (I'm not sure we can enforce it except through code reviews) of extending the *end* of a SQL ENUM, which goes along with never removing one.
2. This wasn't caught by the functional tests, because we don't hit it until we actually `UPDATE` a dataset, so I'm going to add a functional test to publish and unpublish at least one dataset.

Reply:
> there's no hidden internal order-sensitive representation to break.

That's a relief...but, if that's the case, then why is there a `BEFORE` "operator" in that statement?  (Why not just an `ADD`?)

> It's probably best to develop a convention (I'm not sure we can enforce it except through code reviews) of extending the end of a SQL ENUM, which goes along with never removing one.

Makes sense.

> I'm going to add a functional test to publish and unpublish at least one dataset.

💪